src/Sulu/Component/Content/Types/Block/ScheduleBlockVisitor.php

```diff
@@ -0,0 +1,123 @@
+            || !isset($blockPropertyTypeSettings['schedules_enabled'])
+            || !$blockPropertyTypeSettings['schedules_enabled']
+            || !isset($blockPropertyTypeSettings['schedules'])
+            || !\is_array($blockPropertyTypeSettings['schedules'])
+        ) {
+            return $block;
+        }
+
+        $now = $this->requestAnalyzer->getDateTime();
+        $nowTimestamp = $now->getTimestamp();
+
+        $returnBlock = false;
+
+        foreach ($blockPropertyTypeSettings['schedules'] as $schedule) {
+            switch ($schedule['type']) {
+                case 'fixed':
+                    $start = new \DateTime($schedule['start']);
+                    $end = new \DateTime($schedule['end']);
+
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($start->getTimestamp() - $nowTimestamp);
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($end->getTimestamp() - $nowTimestamp);
+
+                    if ($now >= $start && $now <= $end) {
+                        $returnBlock = true;
+                        continue 2;
+                    }
+                    break;
+                case 'weekly':
+                    $year = $now->format('Y');
+                    $month = $now->format('m');
+                    $day = $now->format('d');
+
+                    $start = new \DateTime($schedule['start']);
+                    $start->setDate($year, $month, $day);
+                    $end = new \DateTime($schedule['end']);
+                    $end->setDate($year, $month, $day);
+
+                    if ($end < $start) {
+                        // If the end date is smaller than the start date, it means that the user has entered a time
+                        // combination that spans multiple days. In order to make sure that the start date is before the
+                        // end date the start date has to be set to yesterday.
+                        $start->modify('-1 day');
+                    }
+
+                    $i = 0;
+                    // This loop checks for the coming 7 days for matches of the weekdays selected by the users. This is
+                    // necessary to find the exact weekday on which the next change happens, so that it can be set as
+                    // cachelifetime.
+                    do {
+                        if ($this->matchWeekday($start, $schedule)) {
+                            break;
+                        }
+
+                        $start->modify('+1 day');
+                        $end->modify('+1 day');
+                        ++$i;
+                    } while ($i < 7);
+
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($start->getTimestamp() - $nowTimestamp);
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($end->getTimestamp() - $nowTimestamp);
+
+                    if ($now >= $start && $now <= $end) {
+                        $returnBlock = true;
```

Comment:
dont we need `continue 2` here too? or maybe we can just return as asked above?

Reply:
We could, but since nothing is happen after this `if`, it does not really matter. Would find it even confusing to have a `continue` at the very end :see_no_evil: